File: Algorithms/0462_Minimum_Moves_to_Equal_Array_Elements_II/Java/MinimumMovestoEqualArrayElementsIISolution1.java

// Space: O(1)
// Time: O(n)

class Solution {
    public int minMoves2(int[] nums) {
        if (nums.length <= 1) return 0;
        Arrays.sort(nums);
        int res = 0;
        int mid = nums.length / 2;
        for (int i = 0; i < nums.length; i++) {
            res += Math.abs(nums[i] - nums[mid]);
        }
        return res;
    }
}

File: Algorithms/0462_Minimum_Moves_to_Equal_Array_Elements_II/Python/Minimum_Moves_to_Equal_Array_Elements_II_Solution_1.py

# Space: O(1)
# Time: O(n)

class Solution:

    def minMoves2(self, nums):
        length = len(nums)
        if length <= 1: return 0
        res = 0
        nums = sorted(nums)
        mid = length // 2

        for i in range(length):
            res += abs(nums[i] - nums[mid])

        return res




